### PR TITLE
Add clipboard iframe

### PR DESCRIPTION
**Pull request by mirhamasala**
- Add embed iframe button with transitions  
- Add clipboard feature for iFrame of map

embed_iframe

### CODE REVIEW

File: src/components/TitleMapSelect.vue [hunk 1]
```diff
@@ -60,5 +84,57 @@ export default {
     Legend,
     WorldMap,
   },
+  data() {
+    return {
+      country: null,
+      show: false,
+      copySuccess: null,
+      copyError: null,
```
Comment (yurikoval) [inline src/components/TitleMapSelect.vue:92]: Do we need this? `copyError` is inverse of  `copySuccess`.

File: src/components/TitleMapSelect.vue [hunk 2]
```diff
@@ -60,5 +84,57 @@ export default {
...
+      return false;
+    },
+  },
+  methods: {
+    copyIframe() {
+      const baseUrl = process.env.BASE_URL;
+      let iframeSrc = `${baseUrl}/oembed/`;
+
+      if (this.country) {
+        iframeSrc = `${baseUrl}/oembedmap/${this.country.name.toLowerCase()}.html`;
```
Comment (yurikoval) [inline src/components/TitleMapSelect.vue:122]: iframe url is incorrect and let's use the router api to get the correct one.

File: src/components/TitleMapSelect.vue (hunk 1 above)
Comment (mirhamasala) [reply src/components/TitleMapSelect.vue]: Refactored it.

File: src/components/TitleMapSelect.vue (hunk 2 above)
Comment (mirhamasala) [reply src/components/TitleMapSelect.vue]: Done. Let me know if this is what you had in mind.

File: src/components/TitleMapSelect.vue [hunk 3]
```diff
@@ -60,5 +82,43 @@ export default {
...
+    copyMessage() {
+      if (this.copySuccess) {
+        return 'Copied embed code!';
+      }
+      return 'Copy failed';
+    },
+  },
+  methods: {
+    copyIframe() {
+      const iframeSrc = this.$router.resolve({ name: this.country ? 'CountryOembedMap' : 'OembedMap' }).href;
```
Comment (yurikoval) [inline src/components/TitleMapSelect.vue:109]: iframes will be embedded in other websites, so the url needs to be absolute: `https://www.tourhero.com/en/ultimate-guide-covid19-travel-restrictions-reopening/oembed-map/singapore.html`

Comment (mirhamasala) [reply src/components/TitleMapSelect.vue]: @yurikoval Done

File: src/components/TitleMapSelect.vue [hunk 4]
```diff
@@ -60,5 +82,43 @@ export default {
...
+      if (this.copySuccess) {
+        return 'Copied embed code!';
+      }
+      return 'Copy failed';
+    },
+  },
+  methods: {
+    copyIframe() {
+      const iframeUri = this.$router.resolve({ name: this.country ? 'CountryOembedMap' : 'OembedMap' }).href;
+      return `<iframe src="https://www.tourhero.com${process.env.BASE_URL}${iframeUri.replace('/', '')}"></iframe>`;
```
Comment (yurikoval) [inline src/components/TitleMapSelect.vue:110]: Let's move `iframeUri.replace` out into the line above.

File: vue.config.js [hunk 5]
```diff
@@ -8,7 +8,7 @@ const webpack = require('webpack');
 const Renderer = PrerenderSPAPlugin.PuppeteerRenderer;
 
 const basePath = process.env.NODE_ENV === 'production'
-  ? '/en/ultimate-guide-covid19-travel-restrictions-reopening/' : '';
+  ? '/en/ultimate-guide-covid19-travel-restrictions-reopening/' : '/';
```
Comment (yurikoval) [inline vue.config.js:11]: This change is not necessary -- not related to clipboard.

File: src/views/Country.vue [hunk 6]
```diff
@@ -107,7 +107,7 @@ export default {
       return moment(checkedOn).toISOString();
     },
     articleImage() {
-      return `https://www.tourhero.com${process.env.BASE_URL}img/travel-reopening-cover.jpg`;
+      return `https://www.tourhero.com${this.baseUrl}img/travel-reopening-cover.jpg`;
```
Comment (yurikoval) [inline src/views/Country.vue:110]: Is this related to clipboard?

**Review by yurikoval — CHANGES_REQUESTED** [review]

File: src/views/Country.vue (hunk 6 above)
Comment (mirhamasala) [reply src/views/Country.vue:110]: It's not. I'll remove it.

**Review by yurikoval — APPROVED** [review]

**Pull request merged**